modin/backends/__init__.py

```diff
@@ -13,6 +13,11 @@
 
 from .base import BaseQueryCompiler
 from .pandas import PandasQueryCompiler
-from .pyarrow import PyarrowQueryCompiler
 
-__all__ = ["BaseQueryCompiler", "PandasQueryCompiler", "PyarrowQueryCompiler"]
+__all__ = ["BaseQueryCompiler", "PandasQueryCompiler"]
+try:
```

Comment:
What will happen if we **want** to import `PyarrowQueryCompiler`, but don't have installed pyarrow?
I mean how informing  and explaining error will be?

Reply:
That depends on how you import `PyarrowQueryCompiler`.

Simple `from modin.backends import PyarrowQueryCompiler` would yield `ImportError: cannot import name 'PyarrowQueryCompiler' from 'modin.backends'`, and I think that the error would make developer look into `modin.backends` and he/she would see this `try..except` pattern and then would manually try to do `from modin.backends.pyarrow import PyarrowQueryCompiler` and _then_ would see `cannot import pyarrow`.

So solving the issue would take them one more step compared to empty init, yes, but I think having simpler imports is worth it.

Reply:
We would be able to do lazy submodule imports (which would retain the benefit of simpler imports _and_ emit errors about `pyarrow` at the same time) when we bump our minimal version to 3.7: https://www.python.org/dev/peps/pep-0562/